google-cloud-pubsub/src/main/java/com/google/cloud/pubsub/AbstractSubscriberConnection.java

```diff
@@ -363,50 +264,54 @@ private boolean isRetryable(Status status) {
     }
   }
 
-  private void processReceivedMessages(StreamingPullResponse response) {
-    final List<com.google.pubsub.v1.ReceivedMessage> responseMessages =
-        response.getReceivedMessagesList();
+  protected void setMessageDeadlineSeconds(int messageDeadlineSeconds) {
+    this.messageDeadlineSeconds = messageDeadlineSeconds;
+  }
+
+  protected int getMessageDeadlineSeconds() {
+    return messageDeadlineSeconds;
+  }
+
+  protected void processReceivedMessages(
+      List<com.google.pubsub.v1.ReceivedMessage> responseMessages) {
+    int receivedMessagesCount = responseMessages.size();
+    if (receivedMessagesCount == 0) {
+      return;
+    }
+    Instant now = new Instant(clock.millis());
+    int totalByteCount = 0;
+    final List<AckHandler> ackHandlers = new ArrayList<>(responseMessages.size());
```

Comment:
No we can't, if we are given a blocking executor then merging this loop implies the possibility of delaying the alarm which is certainly not what we want.